source/output_formatter.py

```diff
@@ -0,0 +1,39 @@
+from os import fsync
+from typing import List
+
+from interface import OutputFormatterInterface, OutputData
+
+
+class OutputFormatterFasta(OutputFormatterInterface):
+
+    def write_output(self, output_data: List[OutputData]) -> None:
+
+        for output in output_data:
+
+            (read_id, _, _, _, _, basecalled_seq, _) = output
+
+            if len(basecalled_seq) == 0:
+                return
+
+            print(">%s" % read_id, file=self.output_stream)
+            print(basecalled_seq, file=self.output_stream, flush=True)
+
+        fsync(self.output_stream.fileno())
+
+
+class OutputFormatterFastq(OutputFormatterInterface):
+
+    def write_output(self, output_data: List[OutputData]) -> None:
+
+        for output in output_data:
+
+            (read_id, run_id, read_num, channel_num, start_time, basecalled_seq, quality_scores) = output
```

Comment:
this begs for named tuple

Reply:
`output_data` is a named tuple. But I was thinking that putting it into `print()` function diretly from a named tupple like `output.read_id` would take even more space and be even more ugly. What do you think?

Reply:
Well, for the starter, this does not check whether you do correct naming at destructuring. So it is easy to make a mistake.
Second, you can just rename `output` to `o`  and then it would be more natural.
Finally, you can probably use f-strings instead of format interpolation (assuming python 3) though I guess that is up to discussion whether it will make things nicer or not